transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultPersistor.java

```diff
@@ -339,6 +339,30 @@ public Collection<TransactionOutboxEntry> selectNextInTopics(
     }
   }
 
+  @Override
+  public Collection<TransactionOutboxEntry> selectNextInSelectedTopics(
+      Transaction tx, List<String> topicNames, int batchSize, Instant now) throws Exception {
+    var sql =
+        dialect
+            .getFetchNextInSelectedTopics()
+            .replace("{{table}}", tableName)
+            .replace(
+                "{{topicNames}}",
+                topicNames.stream()
```

Comment:
Need to use parameters here otherwise this is vulnerable to injection attacks and every call will cause a re-evaluation of the query plan, causing a big performance hit.

```
topicNames.stream().map(it-> "?").collect(Collectors.joining(","))
```

Then

```
topicNames.forEach(tn -> stmt.setString(...));
```
```